src/main/java/com/aws/greengrass/secretmanager/SecretManagerService.java

```diff
@@ -87,7 +95,20 @@ protected void install() throws InterruptedException {
     }
 
     private void serviceChanged() {
-        replaceSyncFuture(secretManager::syncFromCloud);
+        Runnable runnable = secretManager::syncFromCloud;
+        replaceSyncFuture(runnable);
+        scheduleNextSync(runnable);
```

Comment:
could we consolidate to just one thread pool? ses with no initial delay?

Reply:
Yep, I tried that but we can't guarantee when the first sync happened during component installation as the `ScheduleFuture` is marked as done/completed only after all the scheduled tasks are done. 
We wait for the initial sync to be completed before marking the component as running.

Reply:
what about keeping track of first time sync using a countdownlatch?